docs/src/doc/decisions-and-ADRs/adr/0008-css-scoping.stories.mdx

```diff
@@ -0,0 +1,33 @@
+import { Meta } from '@storybook/addon-docs/blocks';
+
+<Meta title="Documentation|Decisions and ADRs/ADRs/8) CSS scoping" />
+
+# 8) CSS scoping
+
+Date: 2020-07-24
+
+## Status
+
+Accepted
+
+## Context
+
+CSS authors need to take care to not cause undesired global side effects by creating overly greedy selectors (e.g. styling certain tags on the whole page instead of just the desired ones).
+
+Vue offers a built-in solution for this challenge called [component-scoped CSS](https://vuejs.org/v2/guide/comparison.html#Component-Scoped-CSS). It [adds a data attribute](https://vue-loader.vuejs.org/guide/scoped-css.html#scoped-css) to all component's elements and extends the CSS selectors to match them.
+
+As we are employing base CSS (see ADR #7) and - especially there - care needs to be taken to not accidentally create overly greedy selectors, we must find a solution for scoping it as well.
+
+It is also the case that the developers of this feature team are used to and convinced by the approach of class-based styling as seen in our existing applications (e.g. [termbox](https://gerrit.wikimedia.org/r/plugins/gitiles/wikibase/termbox/+/0d4af4/docs/styles.md), bridge) where we follow the [BEM](http://getbem.com/) methodology which results in a standardized naming scheme for component styles and for example helps debuggability (the classes inspectable in the web browser directly indicate the relevant component).
+
+## Decision
+
+We decided to not go with Vue's component-scoped CSS but instead
+* prefix (e.g. through [prefixwrap](https://www.npmjs.com/package/postcss-prefixwrap)) all our selectors with a dedicated class which scopes the style effects to the matching elements
+* apply styling inside of the components through CSS classes unique across components
+
+## Consequences
+
+Components will be created without the `scoped` attribute on the style section of Single File Components.
+
+As this is a component library and there is no shared `App` DOM node but components will be used in isolation, we will need a way (manual or automatic) to ensure that we consistently add the class matching the "prefix" to all component root elements.
```

Comment:
Writing this helped me realize that this may be a problem.

Reply:
Yes it does for the CSS side. But where does the DOM node with the respective class come from to ensure that the CSS selector even matches?

Imaging prefixwrap + ress in action:
CSS:
```
.wikit p { margin-top: 0 }
```

DOM 1:
```
<div class="wikit-FancyComponent">
   <p>something</p>
</div>
```
Nothing resets the top margin of the `<p>`. Could be fixed by adding the `wikit` class to all components' root node (next to the `wikit-FancyComponent` class).

Worse example:


DOM 2:
```
<p class="wikit-FancyComponent">
   something
</p>
```
The root node _is_ actually the `<p>`. Even adding the `wikit` class there, will not make the `.wikit p` selector match.

In our apps, the node with the equivalent of the `wikit` class (e.g. [in bridge](https://gerrit.wikimedia.org/r/plugins/gitiles/mediawiki/extensions/Wikibase/+/0de341a/client/data-bridge/src/presentation/App.vue#2)), always comes from the application itself, but for our component library we want to be able to use individual components in an otherwise uncontrolled environment.

Maybe I'm just tired and this will go away over the weekend...

Reply:
I agree that this is a problem :/